flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/lsm/sst/SstFile.java

```diff
@@ -0,0 +1,230 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.filestore.lsm.sst;
+
+import org.apache.flink.api.common.serialization.BulkWriter;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.connector.file.src.FileSourceSplit;
+import org.apache.flink.connector.file.src.reader.BulkFormat;
+import org.apache.flink.connector.file.src.util.RecordAndPosition;
+import org.apache.flink.core.fs.FSDataOutputStream;
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.binary.BinaryRowData;
+import org.apache.flink.table.runtime.typeutils.RowDataSerializer;
+import org.apache.flink.table.storage.filestore.KeyValue;
+import org.apache.flink.table.storage.filestore.KeyValueSerializer;
+import org.apache.flink.table.storage.filestore.stats.FieldStatsCollector;
+import org.apache.flink.table.storage.filestore.utils.FileFormat;
+import org.apache.flink.table.storage.filestore.utils.FileUtils;
+import org.apache.flink.table.storage.filestore.utils.RecordIterator;
+import org.apache.flink.table.storage.filestore.utils.RecordReader;
+import org.apache.flink.table.types.logical.RowType;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * This file includes several {@link KeyValue}s, representing a portion of changes applied to the
+ * table.
+ */
+public class SstFile {
+
+    private final Configuration conf;
+    private final RowType keyType;
+    private final RowType valueType;
+
+    private KeyValueSerializer serializer;
+    private KeyValueSerializer previousSerializer;
+    private final RowDataSerializer rowDataSerializer;
+
+    private final BulkFormat<RowData, FileSourceSplit> readerFactory;
+    private final BulkWriter.Factory<RowData> writerFactory;
+
+    public SstFile(Configuration conf, RowType keyType, RowType valueType, FileFormat fileFormat) {
+        this.conf = conf;
+        this.keyType = keyType;
+        this.valueType = valueType;
+
+        this.serializer = new KeyValueSerializer(keyType, valueType);
+        this.previousSerializer = new KeyValueSerializer(keyType, valueType);
+        RowType recordType = KeyValue.schema(keyType, valueType);
+        this.rowDataSerializer = new RowDataSerializer(recordType);
+
+        this.readerFactory = fileFormat.createReaderFactory(recordType);
+        this.writerFactory = fileFormat.createWriterFactory(recordType);
+    }
+
+    public RecordReader<KeyValue> read(Path path) throws IOException {
+        // TODO simplify this logic by using only one serializer and returning java iterator
```

Comment:
We never return `a java iterator`, right? Anyway, we need to return something like batch reader.

Reply:
I'd like to return a closeable iterator here in the future. The iterator will deal with `releaseBatch` and `close` itself and the upper level of file store do not need to see these logic.